### PR TITLE
#2496 change delete work package view to ner form modal

### LINKED ISSUE #2496: DeleteWorkPackageView should use an NERFormModal
### Description

Right now a lot of places use the default Dialog component for a modal instead of the styled NERModal that we've created. This ticket is to replace the use of the MUI dialog with the NERFormModal component in the DeleteWorkPackageView component.

### Acceptance Criteria

- [x]  All use of the dialog component is removed from DeleteWorkPackageView
- [x]  The NERFormModal Component is used instead
- [x]  All functionality is unaffected

### Proposed Solution

See how we've used the NERFormModal elsewhere for examples

### PR DESCRIPTION
**Pull request by vsp05**
## Changes

Changed the Dialog component on the DeleteWorkPackageView to a NERFormModal.

## Test Cases
testing to make sure the functionality still works - i.e. that when you delete a work package, it does, in fact, delete it.
Before deleting WP 0.5.1:
Screenshot 2024-05-30 at 3 50 04 PM
Process of actually deleting WP 0.5.1:
Screenshot 2024-05-30 at 3 50 34 PM
After deleting WP 0.5.1:
Screenshot 2024-05-30 at 3 51 07 PM

second test - making sure that you can't delete a WP unless you enter in the proper name:
what shows up when you try to click the delete button without entering anything into the WBSnum field:
Screenshot 2024-05-30 at 3 52 31 PM
what shows up when you try to click the delete button after entering the wrong WBSnum into the WBSnum fields:
Screenshot 2024-05-30 at 3 53 35 PM

## Screenshots
Full-sized window:
before:
Screenshot 2024-05-30 at 3 56 31 PM

after:
Screenshot 2024-05-30 at 3 55 12 PM

Smallest possible window:
before (you are able to scroll to see the full message):
Screenshot 2024-05-30 at 3 57 09 PM

after (you are able to scroll):
Screenshot 2024-05-30 at 3 55 50 PM

Closes #2496

### CODE REVIEW

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx [hunk 1]
```diff
@@ -40,11 +29,13 @@ const DeleteWorkPackageView: React.FC<DeleteWorkPackageViewProps> = ({ workPacka
...
     control,
-    formState: { errors, isValid }
+    reset,
+    formState: { errors }
   } = useForm({
     resolver: yupResolver(schema),
     defaultValues: {
-      wbsNum: ''
+      wbsNum: '',
+      name: ''
```
Comment (superhvarn) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx:38]: Does name do anything here? If not, you can delete it.

**Review by superhvarn — CHANGES_REQUESTED** [review]
Good work so far!

**Review by superhvarn — DISMISSED** [review]
LGTM!

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx [hunk 2]
```diff
@@ -4,23 +4,12 @@
...
 import { useForm } from 'react-hook-form';
-import NERSuccessButton from '../../../components/NERSuccessButton';
-import NERFailButton from '../../../components/NERFailButton';
 import ReactHookTextField from '../../../components/ReactHookTextField';
 import * as yup from 'yup';
 import { yupResolver } from '@hookform/resolvers/yup';
-import CloseIcon from '@mui/icons-material/Close';
 import { DeleteWorkPackageInputs } from './DeleteWorkPackage';
 
 interface DeleteWorkPackageViewProps {
```
Comment (walker-sean) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx:15]: For the yup schema add a custom message for the required() section saying 'WBS number is required'

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx [hunk 3]
```diff
@@ -54,70 +44,35 @@ const DeleteWorkPackageView: React.FC<DeleteWorkPackageViewProps> = ({ workPacka
...
+      submitText="Delete"
+      handleUseFormSubmit={handleSubmit}
+      onFormSubmit={onSubmitWrapper}
+      formId="delete-wp-form"
+      showCloseButton
+    >
+      <Typography sx={{ marginBottom: '1rem' }}>
+        Are you sure you want to delete Work Package #{wbsPipe(workPackage)}?
+      </Typography>
+      <Typography sx={{ fontWeight: 'bold' }}>This action cannot be undone!!</Typography>
```
Comment (walker-sean) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx:61]: Love the enthusiasm, but one exclamation point is good for this

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx [hunk 4]
```diff
@@ -54,70 +44,35 @@ const DeleteWorkPackageView: React.FC<DeleteWorkPackageViewProps> = ({ workPacka
...
+      <FormControl>
+        <FormLabel sx={{ marginTop: '1rem', marginBottom: '1rem' }}>
+          To confirm deletion, please type in the WBS number of this Work Package.
```
Comment (walker-sean) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx:64]: Nitpick: work package does not need to be capitalized

File: src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx [hunk 5]
```diff
@@ -54,70 +44,35 @@ const DeleteWorkPackageView: React.FC<DeleteWorkPackageViewProps> = ({ workPacka
...
-      <DialogActions>
-        <NERSuccessButton variant="contained" sx={{ mx: 1 }} onClick={onHide}>
-          Cancel
-        </NERSuccessButton>
-        <NERFailButton variant="contained" type="submit" form="delete-wp-form" sx={{ mx: 1 }} disabled={!isValid}>
-          Delete
-        </NERFailButton>
-      </DialogActions>
-    </Dialog>
+    <NERFormModal
```
Comment (walker-sean) [inline src/frontend/src/pages/WorkPackageDetailPage/DeleteWorkPackageModalContainer/DeleteWorkPackageView.tsx:47]: Could you disable the delete button until the wbs number is correct

**Review by walker-sean — CHANGES_REQUESTED** [review]
Nice!

**Review by walker-sean — APPROVED** [review]
LGTM

**Pull request merged**